### ISSUE TITLE
Unable to load model after being saved to a dbfs s3 mount

### ISSUE THREAD

**Issue by antwhite** [mlflow/mlflow#1458]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Databricks 5.4 ML (includes Apache Spark 2.4.3, Scala 2.11)
- **MLflow installed from (source or binary)**:  Databricks
- **MLflow version (run ``mlflow --version``)**: 1.0.0
- **Python version**:  3.6.5
- **npm version (if running the dev UI): -
- **Exact command to reproduce**:  See below for code example

### Describe the problem
When attempting to load a model from S3, the `ArtifactRepository.download_artifacts` method gets stuck in an infinite loop downloading an empty root directory over and over

It appears this is unique to models saved to s3 using DBFS FUSE s3 filesystem. 
I have tracked the issue  down to when the artifacts are listed  in the `S3ArtifactRepository.list_artifacts` method the root directory / key is listed under `Contents` and thus is always returned to be attempted to be downloaded again.

I have been unable to replicate using moto.mock_s3, or using an s3 artefact repository directly.

## Save Model (on databricks)
```python
# mount s3 bucket

ACCESS_KEY = "<aws-access-key>"
SECRET_KEY = "<aws-secret-key>"
ENCODED_SECRET_KEY = SECRET_KEY.replace("/", "%2F")
AWS_BUCKET_NAME = "my-bucket"
MOUNT_NAME = "my-bucket"

dbutils.fs.mount("s3a://{}:{}@{}".format(ACCESS_KEY, ENCODED_SECRET_KEY, AWS_BUCKET_NAME), "/mnt/{}".format(MOUNT_NAME))

from datetime import datetime
from os import path

from keras.layers import Embedding, LSTM, Dense
from keras.models import Sequential

model = Sequential()
model.add(Embedding(2500, 128,input_length = 10, dropout = 0.2))
model.add(LSTM(32, dropout_U = 0.2, dropout_W = 0.2))
model.add(Dense(2,activation='softmax'))
model.compile(loss = 'categorical_crossentropy', optimizer='adam',metrics = ['accuracy'])

# training  omitted

MODEL_VERSION = str.format("{:%Y%m%dT%H%M%S}", datetime.utcnow())
MODEL_NAME = 'model'

model_save_path = path.join('/dbfs/mnt/', MOUNT_NAME, MODEL_VERSION, MODEL_NAME)

mlflow.keras.save_model(model, model_save_path)
```

## Load Model:

```python
AWS_PATH = posixpath.join('s3://', AWS_BUCKET_NAME, MODEL_VERSION, MODEL_NAME)
mlflow.keras.load_model(AWS_PATH)
# Infinite loop
```

**Comment by wardvl-ixor** [comment]
We are having the same problem when downloading a model from s3.  The download_artifacts method keeps downloading an empty root folder.

OSError: [Errno 63] File name too long: '/var/folders/b3/53fzsfxj1_ddkmqplr962s3c0000gn/T/tmpcvgmjaxj/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test/Mlflow_test'

**Comment by RajeshCoditationSystems** [comment]
Do we have solution for this? Getting this error while deploying to sagemaker.

**Comment by sueann** [comment]
Should be fixed in MLflow 1.1 (via #1605). Please re-open if you still see issues. Thanks!
